src/classes/RD2_ElevateIntegrationService.cls

```diff
@@ -0,0 +1,117 @@
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Recurring Donations
+* @description Services related to the RecurringDonation <-> Elevate integration
+*/
+public inherited sharing class RD2_ElevateIntegrationService {
+
+    /**
+    * @description Map of CommitmentId to a specific RecordId (Opp or RD).
+    */
+    private Map<String, Id> commitmentIdToRecordId = new Map<String, Id>();
+
+    /**
+    * @description Pass the map of CommitmentId to RecordId (Opp or RD)
+    * @param commitmentIdToRecordId Map by CommitmentId
+    * @return this
+    */
+    public RD2_ElevateIntegrationService withCommitmentIdMap(Map<String, Id> commitmentIdToRecordId) {
+        this.commitmentIdToRecordId = commitmentIdToRecordId;
+        return this;
+    }
+
+    /**
+    * @description For each CommitmentId, find any existing Opportunities with the matching Commitmentid
+    * and update the RecuringDonation lookup field (only if null).
+    */
+    public void matchRecurringDonationsToOpps() {
+        if (commitmentIdToRecordId == null || commitmentIdToRecordId.isEmpty()) {
+            return;
+        }
+
+        // Skip the process if the Id does not point to a RecurringDonation record
+        if (commitmentIdToRecordId.values()[0].getSobjectType() != npe03__Recurring_Donation__c.SObjectType) {
+            return;
+        }
+
+        List<Opportunity> oppsToUpdate = new List<Opportunity>();
+        for (Opportunity opp : [SELECT Id, npe03__Recurring_Donation__c, CommitmentId__c
+            FROM Opportunity
+            WHERE CommitmentId__c IN :commitmentIdToRecordId.keySet()
+        ]) {
+            if (opp.CommitmentId__c == null || opp.npe03__Recurring_Donation__c != null) {
+                continue;
+            }
+            opp.npe03__Recurring_Donation__c = commitmentIdToRecordId.get(opp.CommitmentId__c);
+            oppsToUpdate.add(opp);
+        }
+
+        if (!oppsToUpdate.isEmpty()) {
+            UTIL_DMLService.updateRecordBatchAndLogErrors(oppsToUpdate, 'RD_Elevate');
```

Comment:
Should we add the context into `ERR_Handler_API.Context`? Is this value used in other teams' code?

Reply:
I was uncertain on this one. The Error Context field/values are not well documented. If anything, they exist for categorization of errors to help in reporting (which assumes anyone actually reports on these). I believe there was some intent to configure error notifications by Context, though that was never fully implemented based on what I can see in the code.

For this I could have gone with the existing `RD` context, which isn't inaccurate.

Looking at the list of context enums, it really feels like the original intent was to tie these to the namespace prefix of the originating class (ADDR, BDI, CONV, HH, RD, etc.). So maybe the right answer here is to create a new enum for RD2 and just use that. I don't necessarily think that adding Elevate to the context will be significantly helpful here. Unless, we decide to create an enum for `Elevate` that truly could be used by other teams. What do you think?

Reply:
Sounds good @force2b , let's use the new enum for `Elevate`.